src/pymodaq_plugins_arduino/daq_viewer_plugins/plugins_0D/daq_0Dviewer_Analog.py

```diff
@@ -0,0 +1,208 @@
+            {'title': 'Channel', 'name': 'ch', 'type': 'led_push', 'value':False, 'label':'On/Off',
+             'tip':'clickto change status, Green: On, Red: Off'},
+            {'title': 'Units:', 'name': 'ai_ch5_units', 'type': 'list',
+             'limits': ['Integer bit','Volts','pH Units', 'Absorbance', 'Transmitance'],
+             'value': 'Volts'},
+        ]}
+
+    ]
+
+    def ini_attributes(self):
+        self.controller: Optional[Arduino] = None
+        pass
+
+    def commit_settings(self, param: Parameter):
+        """Apply the consequences of a change of value in the detector settings
+
+        Parameters
+        ----------
+        param: Parameter
+            A given parameter (within detector_settings) whose value has been changed by the user
+        """
+        ## TODO for your custom plugin
+        if param.name() == "ai0":
+            if param.value():
+                self.controller.set_analog_input(0)
+            else:
+                self.controller.disable_analog_reporting(0)
+        if param.name() == "ai1":
+            if param.value():
+                self.controller.set_analog_input(1)
+            else:
+                self.controller.disable_analog_reporting(1)
+        if param.name() == "ai2":
+            if param.value():
+                self.controller.set_analog_input(2)
+            else:
+                self.controller.disable_analog_reporting(2)
+        if param.name() == "ai3":
+            if param.value():
+                self.controller.set_analog_input(3)
+            else:
+                self.controller.disable_analog_reporting(3)
+        if param.name() == "ai4":
+            if param.value():
+                self.controller.set_analog_input(4)
+            else:
+                self.controller.disable_analog_reporting(4)
+        if param.name() == "ai5":
+            if param.value():
+                self.controller.set_analog_input(5)
+            else:
+                self.controller.disable_analog_reporting(5)
+        # when writing your own plugin replace this line
+
+    #        elif ...
+    ##
+
+    def ini_detector(self, controller=None):
+        """Detector communication initialization
+
+        Parameters
+        ----------
+        controller: (object)
+            custom object of a PyMoDAQ plugin (Slave case). None if only one actuator/detector by controller
+            (Master case)
+
+        Returns
+        -------
+        info: str
+        initialized: bool
+            False if initialization failed otherwise True
+        """
+
+        self.ini_detector_init(slave_controller=controller)
+
+        if self.is_master:
+            self.controller = Arduino(com_port=self.settings['com_port'])  # instantiate you driver with whatever arguments are needed
+
+
+        # TODO for your custom plugin (optional) initialize viewers panel with the future type of data
+        self.dte_signal_temp.emit(DataToExport(name='AnalogInput',
```

Comment:
you don't really have to emit the temp signal here if the type of data emission is the same as the declared plugin (here 0D for 0D data so the viewer will already be correctly set)